monai/transforms/intensity/array.py

```diff
@@ -1137,3 +1131,164 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
         return np.asarray(
             np.interp(img, reference_control_points_scaled, floating_control_points_scaled), dtype=img.dtype
         )
+
+
+class RandGibbsNoise(RandomizableTransform):
+    """
+    Naturalistic image augmentation via Gibbs artifacts. The transform 
+    randomly applies Gibbs noise to 2D/3D MRI images. Gibbs artifacts
+    are one of the common type of type artifacts appearing in MRI scans.
+
+    The transform is applied to all the channels in the data.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+
+
+    Args:
+        prob (float): probability of applying the transform.
+        alpha (float, List(float)): Parametrizes the intensity of the Gibbs noise filter applied. Takes
+            values in the interval [0,1] with alpha = 0 acting as the identity mapping. 
+            If a length-2 list is given as [a,b] then the value of alpha will be 
+            sampled uniformly from the interval [a,b]. 0 <= a <= b <= 1.
+        dim : spatial dimensions of the image (with any leading dimensions). 
+            Set dim = 2 for 2D images, and dim = 3 for 3D images.
+    """
+
+    def __init__(self, prob: float = 0.1, alpha: Union[float, List[float]] = 0.5, dim: int = 3) -> None:
+
+        if type(alpha) == list:
+            if alpha[1] > 1 or alpha[0] < 0:
+                raise AssertionError("alpha must take values in the interval [0,1]")
+            if alpha[0] > alpha[1]:
+                raise AssertionError("When alpha = [a,b] we need a < b.")
+            if len(alpha) != 2:
+                raise AssertionError("If alpha is a list, then its length must be 2.")
```

Comment:
length check would have to come first to avoid index out of bounds error
```suggestion
            if len(alpha) != 2:
                raise AssertionError("If alpha is a list, then its length must be 2.")
            if alpha[1] > 1 or alpha[0] < 0:
                raise AssertionError("alpha must take values in the interval [0,1]")
            if alpha[0] > alpha[1]:
                raise AssertionError("When alpha = [a,b] we need a < b.")
```